hack/run-hive-locally.sh

```diff
@@ -0,0 +1,365 @@
+    # ...
+    local jq_query='to_entries[] | "\(.key)=\(.value)"'
+    local cm
+    local env_value
+    local env
+    local value
+    hl_echo "Getting envFroms from $type_name in ns/$ns"
+
+    while read -r cm; do
+        # Skip ConfigMaps that are already resolved
+        if [ ${resolved_envFrom_cms["$cm"]+_} ]; then
+            hl_echo "cm/$cm already resolved, skipping"
+            continue
+        fi
+
+        # Extract environment variables from ConfigMap
+        hl_echo "Extracting environment variables from cm/$cm in ns/$ns"
+        while read -r env_value; do
+            env="${env_value%=*}"
+            value="${env_value#*=}"
+            resolved_envs["$env"]="$env=$value"
+            hl_echo "Environment variable $env = $value"
+        done < <(oc get "cm/$cm" -n "$ns" -o jsonpath="{.data}" | jq -r "$jq_query")
+
+        # Mark ConfigMap as resolved
+        resolved_envFrom_cms["$cm"]=""
+
+        # Also save ConfigMap.data for future reference
+        oc get "cm/$cm" -n "$ns" -o jsonpath="{.data}" > "./$artifacts_folder/$cm-${hc_spec_generation}.txt"
+    done < <(oc get "$type_name" -n "$ns" -o jsonpath="$jsonpath_query")
+}
+
+function get_incluster_config {
+    local type_names=("deploy/$hive_component_controllers" "sts/$hive_component_clustersync")
+    local type_name
+    for type_name in "${type_names[@]}"; do
+        hl_echo "Getting in-cluster configurations from $type_name in ns/$hive_controllers_ns"
+        get_envs "$type_name" "$hive_controllers_ns"
+        get_envs_from "$type_name" "$hive_controllers_ns"
+    done
+}
+
+function wait_until_hiveconfig_spec_changes {
+    local hiveconfig_spec
+    hiveconfig_spec="$(oc get hiveconfig hive -o jsonpath='{.spec}')"
+    while read -r; do
+        new_hiveconfig_spec="$(oc get hiveconfig hive -o jsonpath='{.spec}')"
+        # As "oc get -w" can give spurious events, here we make sure HiveConfig.spec is actually updated
+        if [[ "$new_hiveconfig_spec" != "$hiveconfig_spec" ]]; then
+            hiveconfig_spec="$new_hiveconfig_spec"
+            break
+        fi
+    done < <(oc get hiveconfig hive --watch-only --output-watch-events)
+    hc_spec_generation=$(( hc_spec_generation + 1 ))
+    hl_echo "Changes to HiveConfig.spec detected"
+    hl_echo "HiveConfig.spec generation = $hc_spec_generation"
+}
+
+function enable_maintenance_mode {
+    hl_echo "Enabling maintenance mode"
+    oc patch hiveconfig hive -p '{"spec": {"maintenanceMode": true}}' --type merge
+
+    hl_echo "Waiting for the controllers to be scaled down"
+    # Blocks until the existing hive-controllers is updated to have 0 replica
+    oc wait -n "$hive_controllers_ns" --for=jsonpath='{.spec.replicas}'=0 deploy/"$hive_component_controllers"
+    # Blocks until the new hive-clustersync is created and has zero replica
+    while ! oc wait -n "$hive_controllers_ns" --for=jsonpath='{.spec.replicas}'=0 sts/"$hive_component_clustersync"; do
+        sleep 1
+    done
```

Comment:
Looks like hive-operator delete and re-create hive-clustersync when necessary (some fields are immutable). So a while loop here to avoid `oc wait` error out.